ECommerceAPI/Endpoints/RoleEndpoints.cs

```diff
@@ -1,6 +1,49 @@
-﻿namespace ECommerceAPI.Endpoints
+﻿using ECommerceAPI.Dtos.Role.Requests;
+using ECommerceAPI.Services.RoleServices.Interfaces;
+namespace ECommerceAPI.Endpoints
 {
     public class RoleEndpoints
     {
+        public static async Task<IResult> CreateRole(IRoleService roleService, RoleRequestDto createRoleRequest)
+        {
+            var response = await roleService.CreateRoleAsync(createRoleRequest);
+            if (response.IsError)
+            {
+                return Results.BadRequest(response.Errors);
+            }
+            return Results.Ok(response.Value);
+        }
+        public static async Task<IResult> GetRoles(IRoleService roleService)
```

Comment:
is not good to pass DI interface as params

Reply:
Yes, I aplogize I felt it was not correcct , I will correct it previously as well